src/components/ButtonSystem/BaseButton.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Not meant to be edited
+ *
+ * This is the base button for all button variations (internal/external links, hash links, non-link buttons)
+ *
+ * Styles/variations are added in MainButtons.style.tsx
+ */
+
+import React, { ComponentProps, DetailedHTMLProps } from 'react';
+import {
+  Link as RouterLink,
+  LinkProps as RouterLinkProps,
+} from 'react-router-dom';
+import { HashLink, HashLinkProps } from 'react-router-hash-link';
+import { assert } from 'common/utils';
+import { BaseMuiButton } from './BaseButton.style';
+import { EventAction, EventCategory, trackEvent } from 'components/Analytics';
+
+type AnchorLinkType = DetailedHTMLProps<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  HTMLAnchorElement
+>;
+
+type StyledButtonProps = ComponentProps<typeof BaseMuiButton>;
+
+type LinkProps = AnchorLinkType | RouterLinkProps | HashLinkProps;
+
+type LinkButtonProps = LinkProps & StyledButtonProps;
+
+const LinkButton: React.FC<LinkButtonProps> = props => {
+  const isLink = props.href || props.to;
+  assert(isLink, "LinkButton needs either 'to' or 'href' as props");
+
+  const isInternalLink = props.to;
+
+  const isHashLink = props.to && props.to.includes('#');
+
+  const { onClick, ...otherProps } = props;
+
+  if (isHashLink) {
+    return (
+      <BaseMuiButton {...otherProps} component={HashLink} onClick={onClick} />
+    );
+  } else if (isInternalLink) {
+    return (
+      <BaseMuiButton {...otherProps} component={RouterLink} onClick={onClick} />
+    );
+  } else {
+    return (
+      <BaseMuiButton
+        {...otherProps}
+        target="_blank"
+        rel="noopener noreferrer"
+        onClick={onClick}
+      />
+    );
+  }
+};
+
+const Button: React.FC<StyledButtonProps> = props => {
+  const hasOnClick = props.onClick;
+  assert(hasOnClick, 'Button needs an onClick');
+
+  const { onClick, ...otherProps } = props;
+
+  return <BaseMuiButton {...otherProps} onClick={onClick} />;
+};
```

Comment:
This can be probably simplified a bit

```tsx
const Button: React.FC<StyledButtonProps> = ({ onClick, ...otherProps }) => {
  assert(onClick, 'Button needs an onClick');
  return <BaseMuiButton {...otherProps} onClick={onClick} />;
}
```